bumble/device.py

```diff
@@ -3382,36 +3406,67 @@ async def update_connection_parameters(
             if l2cap_result != l2cap.L2CAP_CONNECTION_PARAMETERS_ACCEPTED_RESULT:
                 raise ConnectionParameterUpdateError(l2cap_result)
 
-        result = await self.send_command(
-            HCI_LE_Connection_Update_Command(
-                connection_handle=connection.handle,
-                connection_interval_min=connection_interval_min,
-                connection_interval_max=connection_interval_max,
-                max_latency=max_latency,
-                supervision_timeout=supervision_timeout,
-                min_ce_length=min_ce_length,
-                max_ce_length=max_ce_length,
+        with closing(EventWatcher()) as watcher:
```

Comment:
suggestion: move the call to `self.send_command()` to a local function, which the code followibng `with closing(...) ...:` can invoke, but can also be invoked without the `with` close when `wait_for_completion` is `False`, so that we don't unnecessarily create the context and the future in that case.